File: Frontend/SPFrontend/src/app/components/models/CorrectionDTO.ts
import {EvaluationCriteriaDTO} from "./EvaluationCriteriaDTO";

export class CorrectionDTO {

  questionId: number;
  evaluationCriteria: EvaluationCriteriaDTO;
  matrNr: number;
  comment: string;
  reachedPoints: number;
  status: string;
  examId: number;

  constructor(questionId: number, evaluationCriteria: EvaluationCriteriaDTO, matrNr: number, comment: string, reachedPoints: number, status: string, examId: number) {
    this.questionId = questionId;
    this.evaluationCriteria = evaluationCriteria;
    this.matrNr = matrNr;
    this.comment = comment;
    this.reachedPoints = reachedPoints;
    this.status = status;
    this.examId = examId;
  }
}

File: Backend/src/main/java/de/hft_stuttgart/winf/proj2/sp/backend/dto/CorrectionDTO.java
package de.hft_stuttgart.winf.proj2.sp.backend.dto;

import de.hft_stuttgart.winf.proj2.sp.backend.util.Column;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

@AllArgsConstructor
@Getter
@ToString(callSuper = true)
@EqualsAndHashCode
public class CorrectionDTO {

    @Column("question_id")
    private Integer questionId;
    @Column(value="evaluation_criteria", isObject = true)
    private QuestionCriteriaDTO evaluationCriteria;
    @Column("matr_nr")
    private Integer matrNr;
    @Column("comments")
    private String comment;
    @Column("reached_points")
    private Float reachedPoints;
    @Column("status")
    private String status;
    @Column("exam_id")
    private Integer examId;

    public CorrectionDTO(){}
}
